source/common/config/well_known_names.h

```diff
@@ -36,23 +57,42 @@ typedef ConstSingleton<NetworkFilterNameValues> NetworkFilterNames;
 class HttpFilterNameValues {
 public:
   // Buffer filter
-  const std::string BUFFER = "buffer";
+  const std::string BUFFER = "envoy.buffer";
   // Dynamo filter
-  const std::string DYNAMO = "http_dynamo_filter";
+  const std::string DYNAMO = "envoy.http_dynamo_filter";
   // Fault filter
-  const std::string FAULT = "fault";
+  const std::string FAULT = "envoy.fault";
   // GRPC http1 bridge filter
-  const std::string GRPC_HTTP1_BRIDGE = "grpc_http1_bridge";
+  const std::string GRPC_HTTP1_BRIDGE = "envoy.grpc_http1_bridge";
   // GRPC json transcoder filter
-  const std::string GRPC_JSON_TRANSCODER = "grpc_json_transcoder";
+  const std::string GRPC_JSON_TRANSCODER = "envoy.grpc_json_transcoder";
   // GRPC web filter
-  const std::string GRPC_WEB = "grpc_web";
+  const std::string GRPC_WEB = "envoy.grpc_web";
   // IP tagging filter
-  const std::string IP_TAGGING = "ip_tagging";
+  const std::string IP_TAGGING = "envoy.ip_tagging";
   // Rate limit filter
-  const std::string RATE_LIMIT = "rate_limit";
+  const std::string RATE_LIMIT = "envoy.rate_limit";
   // Router filter
-  const std::string ROUTER = "router";
+  const std::string ROUTER = "envoy.router";
+
+  HttpFilterNameValues() {
+    const std::vector<std::string> prefixed_v2_names = {
```

Comment:
This thing seems fragile to me but I can't really think of a better way to do it. Can we potentially at least share this code with above via some small helper class that takes an array of strings as part of the constructor?

Reply:
Yeah, my bad.  Should've been done that way in the first place.